### PR TITLE
Fixed #92 (.xls exporting)

### LINKED ISSUE #92: Excel Export Issues
Volunteer export - add organization column.
Request export - missing field header, columns misaligned.

### PR DESCRIPTION
**Pull request by stavlocker**

### CODE REVIEW

File: client/forms.py [hunk 1]
```diff
@@ -71,7 +71,6 @@ class VolunteerForm(forms.Form):
     notes = forms.CharField(max_length=DEFAULT_MAX_FIELD_LENGTH, required=False)
     transportation = forms.ChoiceField(choices=MOVING_WAYS)
     hearing_way = forms.ChoiceField(choices=HEARING_WAYS)
-    area = forms.MultipleChoiceField(choices=AREAS, widget=forms.CheckboxSelectMultiple())
```
Comment (stavlocker) [inline client/forms.py]: This line appeared twice, hence the deletion

File: client/models.py [hunk 2]
```diff
@@ -117,6 +116,13 @@ def get_active_certificates(self):
     hearing_way = models.CharField(max_length=SHORT_FIELD_LENGTH, choices=HEARING_WAYS)
     schedule = models.OneToOneField(VolunteerSchedule, on_delete=models.CASCADE, null=True)
 
+    @property
+    def full_name(self):
+        return f'{self.first_name} {self.last_name}'
+
```
Comment (itay-sho) [inline client/models.py]: this is brilliant

File: server/xls_exporter.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                                the text to display in the header row
+    :param filename: The filename to generate without the .xls extension
+                     Default: <model_name_lower>_data-%Y_%m_%d-%H%M%S (with formatted time)
+    :param spreadsheet_name: The name of the spreadsheet in the file
+                             Default: <model_name>s
+
+    :return: HttpResponse of content type application/ms-excel with the .xls file as an attachment
+    """
+    response = HttpResponse(content_type='application/ms-excel')
+    response['Content-Disposition'] = 'attachment; filename="{}.xls"'.format(
```
Comment (itay-sho) [inline server/xls_exporter.py]: use f-strings here
https://www.python.org/dev/peps/pep-0498/

File: server/xls_exporter.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        filename if filename is not None else '{}_data-{}'.format(
+            model_type.__name__.lower(),
+            datetime.datetime.now().strftime('%Y_%m_%d-%H%M%S')
+        )
+    )
```
Comment (itay-sho) [inline server/xls_exporter.py]: instead of a one liner do: if filename is None and assign the filename as you wish

File: server/xls_exporter.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+            if isinstance(value, bool):
+                value = 'כן' if value else 'לא'
+            elif isinstance(value, datetime.datetime):
+                value = value.strftime('%Y/%m/%d %H:%M')
+            elif isinstance(value, django.db.models.manager.Manager):
+                value = ', '.join([str(item) for item in value.all()])
+            elif isinstance(value, models.Model):
+                value = str(value)
+
+            ws.write(row_num, col_num, value)
```
Comment (itay-sho) [inline server/xls_exporter.py]: super neat.

File: server/xls_exporter.py (hunk 3 above)
Comment (stavlocker) [reply server/xls_exporter.py]: resolved

File: server/xls_exporter.py (hunk 4 above)
Comment (stavlocker) [reply server/xls_exporter.py]: you're right. resolved

**Comment by stavlocker** [comment]
@itay-sho Fixed CR (I'm still trying to figure out how to use the CR system in GitHub)

File: server/xls_exporter.py [hunk 6]
```diff
@@ -0,0 +1,68 @@
...
+    """
+    if filename is None:
+        current_time = datetime.datetime.now().strftime('%Y_%m_%d-%H%M%S')
+        filename = f'{model_type.__name__.lower()}_data-{current_time}'
+
+    response = HttpResponse(content_type='application/ms-excel')
+    response['Content-Disposition'] = f'attachment; filename="{filename}.xls"'
+
+    wb = xlwt.Workbook(encoding='utf-8')
+    ws = wb.add_sheet(spreadsheet_name if spreadsheet_name is not None else '{}s'.format(model_type.__name__))
```
Comment (itay-sho) [inline server/xls_exporter.py]: f-strings

Comment (itay-sho) [reply server/xls_exporter.py]: btw, is this your easy way to find plurals of stuff?
I'm pretty sure that django has a really good implementation for that that you may want to use

**Comment by stavlocker** [comment]
@itay-sho Done :)

File: server/views.py [hunk 7]
```diff
@@ -394,39 +395,30 @@ def find_closes_persons(request, pk):
...
-    rows = Volunteer.objects.all().values_list('full_name', 'first_name', 'last_name', 'tz_number', 'volunteer_type',
-                                               'age',
-                                               'phone_number', 'areas', 'city', 'email', 'available_saturday',
-                                               'guiding', 'notes')
-    for row in rows:
-        row_num += 1
-        for col_num in range(len(row)):
-            ws.write(row_num, col_num, row[col_num], font_style)
+    fields_descriptions = {
+        'first_name': 'שם פרטי',
```
Comment (astar26) [inline server/views.py:399]: Add volunteer ID, too

Comment (stavlocker) [reply server/views.py:400]: resolved

**Comment by stavlocker** [comment]
@astar26 Done.
Regarding the filtering, you mean a filtered excel file, right?
We could make the button export the data currently being filtered, that sounds like a good idea.
Should I open an issue?

**Comment by astar26** [comment]
> 
> 
> @astar26 Done.
> Regarding the filtering, you mean a filtered excel file, right?
> We could make the button export the data currently being filtered, that sounds like a good idea.
> Should I open an issue?

Yes, open the issue, mention abstraction of volunteer querying and filtering as part of it (seems to me like shared code, mostly)

**Comment by stavlocker** [comment]
@astar26 @itay-sho You both left comments which I resolved, but you didn't mark as resolved, so GitHub thinks there are still changes needed here.
If there aren't any additional comments can one of you merge to develop?

**Comment by itay-sho** [comment]
you have some merge conflicts with develop? can you resolve them? after that you'll get the approval :)

**Comment by stavlocker** [comment]
@itay-sho Resolved conflicts.

**Pull request merged**